README.md

```diff
@@ -4,10 +4,23 @@
 
 **Supported Swift Versions:** Swift 4.1, Swift 4.2
 
+ReSwift-Thunk allows you to replace dispatching actions with dispatching blocks that perform synchronous or asynchronous work before they dispatch.
+
 When [ReSwift](https://github.com/ReSwift/ReSwift/) is a [Redux](https://github.com/reactjs/redux)-like implementation of the unidirectional data flow architecture in Swift, ReSwift-Thunk is like [redux-thunk](https://github.com/reduxjs/redux-thunk). 
 
 ## Why Use ReSwift-Thunk?
 
+In ReSwift, you should not write complex stuff in your reducers, but might be tempted to do so nevertheless. Where else should network requests go? Since reducers have access to the current state _and_ the incoming action, they seem promising at first. But reducers should be dumb and perform their work quickly to keep your app performant. So where else would you process complex actions?
+
+Enter thunks.
+
+1. **Save time and boilerplate code.** Some use cases involve complex asynchronous processes that are a pain to write with plain ReSwift. Think about changing a file from your app. Without thunks, a user interaction would have to (1) dispatch the change action to the store, then (2) a reducer would have to enqueue the change into the app state as a pending command. (3) The processor subscribes to the queue of pending file changes and performs the actual write to the disk, (4) dispatching a completion action which (5) get reduced to dequeuing the command from the app state. -- In some cases, you can get rid of all of this with a thunk that processes the file change. You can handle network requests in a similar fashion. Not all asynchronous requests should be treated this way, since you will want your app to be aware of some "waiting for server response" states, e.g. clearing a table view and displaying a loading indicator. But in case you don't need that, thunks are your friend!
```

Comment:
I think this three points are too opinionated and can also be misleading. For instance, what you say here is not correct, as you are not going to save any boilerplate code using thunks if you want to accomplish the same than without them. If you want to store and react to that intermediate info, you'll need to do all the dance dispatching the several actions. Also, if you want to clear a table view or display a loading indicator locally (ie, without saving that info in your store's state), you can as well even using a thunk.

IMO, the main selling point of a thunk is the ability to colocate an asynchronous operation (ie, a one that involves side effects) in the same place it is triggered, without losing reusability and with the ergonomics of being able to treat it as a normal action. This makes it easier to test it as well, as you can mock its dependencies or use DI and test the thunk as a unit.

What I mean is that if a view controller, for example, you have access to your API manager class and your store, you can build a _thunk_ using a simple instance function that encapsulates the asynchronous behaviour, writing the same amount of code, but it will be more couple to that controller and therefore more difficult to test.